flink-connectors/flink-connector-hive/src/main/java/org/apache/flink/table/catalog/hive/HiveCatalogBase.java

```diff
@@ -407,4 +443,201 @@ private Table getHiveTable(ObjectPath tablePath) throws TableNotExistException {
 				String.format("Failed to get table %s from Hive metastore", tablePath.getFullName()), e);
 		}
 	}
+
+	// ------ partitions ------
+
+	@Override
+	public boolean partitionExists(ObjectPath tablePath, CatalogPartitionSpec partitionSpec)
+		throws CatalogException {
+		try {
+			Table hiveTable = getHiveTable(tablePath);
+			return client.getPartition(tablePath.getDatabaseName(), tablePath.getObjectName(),
+				getFullPartitionValues(tablePath, partitionSpec, getFieldNames(hiveTable.getPartitionKeys()))) != null;
+		} catch (NoSuchObjectException | TableNotExistException | PartitionSpecInvalidException e) {
+			return false;
+		} catch (TException e) {
+			throw new CatalogException(
+				String.format("Failed to get partition %s of table %s", partitionSpec, tablePath), e);
+		}
+	}
+
+	@Override
+	public void createPartition(ObjectPath tablePath, CatalogPartitionSpec partitionSpec, CatalogPartition partition, boolean ignoreIfExists)
+		throws TableNotExistException, TableNotPartitionedException, PartitionSpecInvalidException, PartitionAlreadyExistsException, CatalogException {
+
+		validateCatalogPartition(partition);
+
+		Table hiveTable = getHiveTable(tablePath);
+
+		ensurePartitionedTable(tablePath, hiveTable);
+
+		try {
+			client.add_partition(createHivePartition(hiveTable, partitionSpec, partition));
+		} catch (AlreadyExistsException e) {
+			if (!ignoreIfExists) {
+				throw new PartitionAlreadyExistsException(catalogName, tablePath, partitionSpec);
+			}
+		} catch (TException e) {
+			throw new CatalogException(
+				String.format("Failed to create partition %s of table %s", partitionSpec, tablePath));
+		}
+	}
+
+	@Override
+	public void dropPartition(ObjectPath tablePath, CatalogPartitionSpec partitionSpec, boolean ignoreIfNotExists)
+		throws PartitionNotExistException, CatalogException {
+		try {
+			Table hiveTable = getHiveTable(tablePath);
+			ensurePartitionedTable(tablePath, hiveTable);
+			client.dropPartition(tablePath.getDatabaseName(), tablePath.getObjectName(),
```

Comment:
Can we skip line #490 - 491 and let client.dropPartition() to take care of exceptions? This may avoid additional trips to Hive metastore.

Reply:
Yeah I guess we can do this for `dropPartition`, because `dropPartition` doesn't require a precise exception like `createPartition`. So it's probably OK to wrap anything we get in a `PartitionNotExistException`

Reply:
getHiveTable() requires a round trip. Compared to precise exception msg, I'd say we minimize the number of trips.

Reply:
We need to get the hive table in order to get the partition columns (which in turn is used to get ordered partition values). So I don't think we can get rid of the trip to HMS